src/database/migrations/20221025150522_create_fill_event_table.ts

```diff
@@ -0,0 +1,41 @@
+import { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+  await knex.schema.createTable('fill_event', (table) => {
+    table.increments('id');
+    table.uuid('user').references('id').inTable('user').notNullable();
+    table
+      .uuid('cylinder_set')
+      .references('id')
+      .inTable('diving_cylinder_set')
+      .notNullable();
+    table.string('info', 2048); // arbitrary number
+    table.timestamps(true, true);
+  });
+
+  await knex.schema.createTable('fill_cylinder_event', (table) => {
+    table
+      .increments('fill_event')
+      .references('id')
+      .inTable('fill_event')
+      .notNullable();
+    table
+      .uuid('cylinder')
+      .references('id')
+      .inTable('diving_cylinder')
+      .notNullable();
+    table
+      .string('gas', 15)
+      .checkIn(['pressurized_air', 'oxygen', 'helium', 'argon'])
```

Comment:
```suggestion
      .checkIn(['air', 'oxygen', 'helium', 'argon'])
```
Ja jos halutaan käyttää jotain muuta termiä kuin  `air` niin sitten varmaan `compressed_air`

Reply:
Jaaa tälläinen rakennehan kuuluisi kait olla enum ([doc](https://mariadb.com/kb/en/enum/) ja [doc](https://knexjs.org/guide/schema-builder.html#enum-enu)). Muistaen että pitää tarkistaa sql_mode == strict ettei dataintegriteetti vaarannu tyhjistä enum-arvoista.

Reply:
Ja noita kaasujahan on siinä pullotäytössä aina jokin osapaine, tälläinen sarake antaisi täyttää vain yhtä kaasua pulloon. Todellisuudessa pullosettiin päätyy jotain seuraavista kaasuseoksista:
* ilma
* nitrox
* trimix
* heliox
* argon
* lista varmaan jatkuu

Se että halutaanko tämä tieto tälläisenään kantaan on toinen kysymys.

Reply:
Jos ja kun halutaan tallentaa tieto siitä mitä kaasuja täytössä on käytetty ja kuinka paljon niin tätä ongelmaa pitää lähestyä jotenkin toisin.

Reply:
Yksi vaihtoehto olisi varmaa se, että käytettäville kaasuille (ilma, happi, helium, argon, diluentti) olisi jokaiselle oma sarakkeensa johon arvoksi tulisi osapaine bareina. Tai sitten kirjataan maksullisten kaasujen osalta paljonko minkäkin varastopullon paine on laskenut (tämä muodostaa täytön hinnan). Tai sitten käyttöliittymässä on erikseen kentät varastopullojen paineille ja tämän taulun hintasarakkeen arvo lasketaan niistä?. Tai sekä että. En tiedä. En ole (vielä) kaasunsekoituksen asiantuntija.

Harmi että asiantuntijamme on lomalla, meille luvatut speksit jumissa bittiavaruudessa ja tiukempien kaasuseoksien kirjaamiseen käytetyn taulukon kuvat ottamatta.

Reply:
Kyllähän se harmittaa itse kutakin.
Miulla on nyt idea kolmanteen iteraatioon tämän pohjalta